### ISSUE TITLE
dispose parameter is not built now

### ISSUE THREAD

**Issue by dan-leech** [Milad-Akarie/injectable#240]
```Dart
@module
abstract class ComputerModule {
  @LazySingleton(dispose: disposeComputer)
  Computer get computer {
    return Computer();
  }
}

void disposeComputer(Computer computer) {
  computer.stop();
}
```

I get build error:
```
[SEVERE] injectable_generator:injectable_builder on lib/di/modules/computer.dart:

Can not resolve function type 
Try using an alias e.g typedef MyFunction = void Function(Computer);
```

version 1.4.0 works ok

**Comment by Milad-Akarie** [comment]
@dan-leech This should be fixed in injectable_generator 1.5.0 
please reopen if it's not.

**Comment by zainulhassan815** [comment]
```dart
import 'dart:async';

import 'package:cloud_firestore/cloud_firestore.dart';
import 'package:ggc_science_fair/auth/auth.dart';
import 'package:ggc_science_fair/projects/projects.dart';
import 'package:ggc_science_fair/projects/repository/remote_project_repository.dart';
import 'package:injectable/injectable.dart';

@module
abstract class RepositoryModule {
  FutureOr disposeDataSource(ProjectsRepository instance) {
    instance.dispose();
  }

  @LazySingleton(
    as: ProjectsRepository,
    dispose: disposeDataSource, // Doesn't work. Throws constant expression error.
    dispose: (ProjectsRepository repo) => repo.dispose(), // This also doesn't work.
  )
  RemoteProjectsRepository projectsRepository(
    FirebaseFirestore firestore,
    UserRepository userRepository,
  ) {
    return RemoteProjectsRepository(
      firestore: firestore,
      userRepository: userRepository,
    );
  }
}

```
![](https://drive.google.com/uc?export=download&id=1Y-wFctQz_50saHIO4RzkA22mfWEM9m-_)

I get this error if I use a higher order function or a lambda function. But if I annotate the class's dispose method with `[@disposeMethod](https://pub.dev/packages/injectable#disposing-of-singletons)` annotation, everything works. Is it intended to do so? If not how can I fix this as I want to keep dependency inject in this module.
